src/odemis/odemisd/main.py

```diff
@@ -90,40 +92,54 @@ def __init__(self, model_file, settings_file, create_sub_containers=False,
             logging.exception("When instantiating file %s", self._model.name)
             raise IOError("Unexpected error at instantiation")
 
-    def _on_va_change(self, _):
+    def _observe_persistent_va(self, comp, prop_name):
         """
-        Callback to VA change, calls write_persistent_data
+        Listen to changes in persistent va, update the value of instantiator.persistent_props
+        and write data to file.
+        comp (HwComponent): component
+        prop_name (str): name of va
         """
-        self._write_persistent_data()
+        def on_va_change(value, comp=comp.name, prop=prop_name):
```

Comment:
To avoid confusion, let's name these variables: `comp_name=comp.name, prop_name=prop_name`.